src/cli/utils/completions.go

```diff
@@ -0,0 +1,194 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/alcionai/corso/src/pkg/backup"
+	"github.com/alcionai/corso/src/pkg/fault"
+	"github.com/alcionai/corso/src/pkg/path"
+	"github.com/alcionai/corso/src/pkg/services/m365"
+	"github.com/alcionai/corso/src/pkg/store"
+)
+
+func GetBackups(ctx context.Context, cmd *cobra.Command, service path.ServiceType) ([]*backup.Backup, error) {
+	r, _, err := GetAccountAndConnect(ctx, cmd, service)
+	if err != nil {
+		return nil, err
+	}
+
+	defer CloseRepo(ctx, r)
+
+	return r.BackupsByTag(ctx, store.Service(service))
+}
+
+type completionFunc func(*cobra.Command, []string, string) ([]string, cobra.ShellCompDirective)
+
+func BackupIDCompletionFunc(service path.ServiceType) completionFunc {
+	return func(
+		cmd *cobra.Command,
+		args []string,
+		toComplete string,
+	) ([]string, cobra.ShellCompDirective) {
+		bs, err := GetBackups(cmd.Context(), cmd, service)
+		if err != nil {
+			return cobra.AppendActiveHelp(
+				nil,
+				fmt.Sprintf("Unable to fetch %s backups", service)), cobra.ShellCompDirectiveNoFileComp
+		}
+
+		if len(bs) == 0 {
+			return cobra.AppendActiveHelp(
+				nil,
+				fmt.Sprintf("No %s backups found", service)), cobra.ShellCompDirectiveNoFileComp
+		}
+
+		backups := make([]string, len(bs))
+		for _, b := range bs {
+			backups = append(backups, fmt.Sprintf("%s\tCreated at %s", b.GetID(), b.CreationTime))
+		}
+
+		return cobra.AppendActiveHelp(backups, "Choose backup ID to use"), cobra.ShellCompDirectiveNoFileComp
+	}
+}
+
+func UsersCompletionFunc(service path.ServiceType) completionFunc {
```

Comment:
Could you please add some description for these functions? Also, would it be possible to add tests for these completion funcs?